docs/content/docs/build/container-states-events.md

```diff
@@ -0,0 +1,336 @@
+
+{{< callout note >}}
+
+In the Fluid APIs, the terms "attach", "attaching", "attached" and "detached" mean publish, publishing, published, and unpublished, respectively.
+
+{{< /callout >}}
+
+#### Published to the service
+
+When publishing completes, the container is **published**, so it exists in the Fluid service as well as the creating client. 
+Subsequent clients can [connect to the container](./containers.md#connecting-to-a-container), synchronize with its latest data values, and edit them.
+
+A container can never transition back to an unpublished state, but it can transition to a deleted state. 
+
+#### Deleted from the service
+
+A container transitions to **deleted** state when it is deleted. 
+Any connected clients become disconnected (although they may have been already disconnected before the deletion; see [Deleting a container from the service](#deleting-a-container-from-the-service)), but the container is not disposed on the clients. 
+Clients could continue to edit the container's data, but the data isn't shared anymore. Each client's changes are known only to it. 
+
+##### Deleting a container from the service
+
+Deleting a container is a service-specific feature, so you should refer to the documentation associated with the specific Fluid service you are using for instructions on how to do it. 
+To prevent clients from being suddenly cut off from data sharing, the service may put restrictions on when a container can be deleted. 
+For example, it may allow deletion only if all clients have been disconnected from the container for a specified period of time. 
+
+Once it is deleted, the container cannot move back to the **published** state again.
+It is possible for a client to create a new container using the same schema as a deleted container. 
+If a record of the final data values of the deleted container are available to the creating client, they can be used to initialize the new container. 
+But the new container has a different ID from the deleted one and subsequent clients will have to connect to it using the new ID.
+
+#### Handling publication status
+
+Your code can test for the publication status with the [container.AttachState]({{< relref "/docs/apis/fluid-static/fluidcontainer-class.md#attachstate-property" >}}) property which has an [AttachState]({{< relref "/docs/apis/container-definitions.md#attachstate-enum" >}}) value. 
+This can be useful if your application will publish the container in some code paths, but not others. 
+For example, in complex cases, you may need to determine whether the container has already been published before you call `container.attach`. The following is a simple example.
+
+```typescript
+if (container.attachState !== AttachState.Attached) {
+    container.attach();
+}
+```
```

Comment:
I don't think this is the best example. Two clients with detached/unpublished containers will always be able to separately call `attach()` on their containers and succeed, but that will result in two separate backing documents in the service. Before a container is attached, there is no concept of several clients sharing it. The APIs are also designed in a way where this scenario doesn't arise. If a client wants to create a new container, they'll use the "create new" flow (`loader.createDetachedContainer()`), and if a client already has a "container" (document) id, it'll use the "load existing flow" (`loader.loadExisting(id)`).

A better example might be logic that should only run in the creating client *before* a container is attached/published. Every client that loads that "container" (the document that backs it) might have to go through a code path like this:

```typescript
if (container.attachState !== AttachState.Attached) {
  // First-time initialization that only runs in the creating client
}
```

In all subsequent containers, `container.attachState === AttachState.Attached` so they'll skip that code.

Note this also means that subsequent clients will never see `AttachState.Detached` nor `AttachState.Attaching`.

Reply:
> I don't think this is the best example. Two clients with detached/unpublished containers will always be able to separately call `attach()` on their containers and succeed, but that will result in two separate backing documents in the service. Before a container is attached, there is no concept of several clients sharing it. The APIs are also designed in a way where this scenario doesn't arise. If a client wants to create a new container, they'll use the "create new" flow (`loader.createDetachedContainer()`), and if a client already has a "container" (document) id, it'll use the "load existing flow" (`loader.loadExisting(id)`).
> 
> A better example might be logic that should only run in the creating client _before_ a container is attached/published. Every client that loads that "container" (the document that backs it) might have to go through a code path like this:
> 
> ```ts
> if (container.attachState !== AttachState.Attached) {
>   // First-time initialization that only runs in the creating client
> }
> ```
> 
> In all subsequent containers, `container.attachState === AttachState.Attached` so they'll skip that code.
> 
> Note this also means that subsequent clients will never see `AttachState.Detached` nor `AttachState.Attaching`.

@alexvy86 Actually, your suggestion is exactly what I intended. I'll try to clarify this passage.